src/openrct2-ui/UiContext.cpp

```diff
@@ -669,7 +669,7 @@ class UiContext final : public IUiContext
             SDLException::Throw("SDL_CreateWindow(...)");
         }
 
-        ToggleScreensaverLock();
+        ToggleScreenSaverLock();
```

Comment:
'Toggle' suggests that it's changing the lock, but this function is only applying the setting, right?

Reply:
It actually changes the state of the lock. If you begin with enabled it will change setting to disabled and so on.

Reply:
`gConfigGeneral.disable_screensaver` is read but not changed. It also wouldn't really make sense to toggle the setting when opening the window.

Reply:
`gConfigGeneral.disable_screensaver` is changed in lines 642 - 647 of `src/openrct2-ui/windows/Options.cpp`:
```
        case WIDX_DISABLE_SCREENSAVER_LOCK:
            gConfigGeneral.disable_screensaver ^= 1;
            ToggleScreenSaverLock();
            config_save_default();
            w->Invalidate();
            break;
```
I followed also a suggestion by @AaronVanGeffen to export this logic to new method and invoke it in relevant places.

The problem here (as I understand it) is that SDL by default inhibits screensaver and monitor power saving.
Leaving this function only in options window logic will cause to read this value and change related setting only if someone goes to options window and (un)checks the checkbox.
Without this, independently of what is saved in config.ini file, this setting won't be applied to the game at the start and thus screensaver will be always inhibited.

Reply:
I think @Broxzier is suggesting the name `ToggleScreenSaverLock` implies a setting change, rather than applying the current setting. In that sense, I agree something like `ApplyScreenSaverLock` might be more appropriate. However, at the same time, SDL is still toggling the screensaver inhibition status. I therefore feel a name change is not necessarily needed.

Reply:
> However, at the same time, SDL is still toggling the screensaver inhibition status.

That doesn't sound right. Shouldn't this be called on startup already rather than when the options window gets opened?

>  I therefore feel a name change is not necessarily needed.

The function calls the corresponding SDL function, the toggling happens in the user input handler. How I see it, 'toggle' doesn't make sense here.

Reply:
> That doesn't sound right. Shouldn't this be called on startup already rather than when the options window gets opened?

Perhaps there is a misunderstanding here. It is called when the game starts: in `UiContext`'s `CreateWindow`. The only time it's called in the Options window, is when you click the relevant checkbox, in order to change the state.

Reply:
> Perhaps there is a misunderstanding here. It is called when the game starts: in UiContext's CreateWindow. The only time it's called in the Options window, is when you click the relevant checkbox, in order to change the state.

Ah! That explains the confusion, I was indeed thinking these changes were made in the options window.

> Again, I don't feel strongly about this. If you think it needs changing, let's.

Alrighty. It's cleaner imo, identifiers should reflect what they do, not how they are used.

Reply:
> Alrighty. It's cleaner imo, identifiers should reflect what they do, not how they are used.
Looking from that perspective maybe renaming to `ApplyScreenSaverLockStatusChange` should be more descriptive than Toggle.